index.js

```diff
@@ -165,20 +165,25 @@ module.exports = class Hyperswarm extends EventEmitter {
     conn.on('error', noop)
     conn.on('open', () => {
       opened = true
-      this._connectDone()
       this.connections.add(conn)
+      this._connectDone()
       conn.removeListener('error', noop)
       peerInfo._connected()
       peerInfo.client = true
       this.emit('connection', conn, peerInfo)
       this._flushMaybe(peerInfo)
     })
+
+    this.connecting++
```

Comment:
Should keep this line above where it was before? And just leave the emit in the bottom of the func?